src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -916,93 +936,92 @@ def call(
-            output_hidden_states = inputs.get("output_hidden_states", output_hidden_states)
-        else:
-            input_ids = inputs
-
-        use_cache = use_cache if use_cache is not None else self.config.use_cache
-        if decoder_input_ids is None:  # Classification
+        inputs = input_processing(
+            func=self.call,
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            decoder_input_ids=decoder_input_ids,
+            decoder_attention_mask=decoder_attention_mask,
+            encoder_outputs=encoder_outputs,
+            past_key_values=past_key_values,
+            use_cache=use_cache,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            training=training,
+            kwargs_call=kwargs,
+        )
+        use_cache = inputs["use_cache"] if inputs["use_cache"] is not None else self.config.use_cache
+        if inputs["decoder_input_ids"] is None:  # Classification
             use_cache = False
-        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
-        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_attentions = (
+            inputs["output_attentions"] if inputs["output_attentions"] is not None else self.config.output_attentions
+        )
         output_hidden_states = (
-            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+            inputs["output_hidden_states"]
+            if inputs["output_hidden_states"] is not None
+            else self.config.output_hidden_states
         )
+        return_dict = inputs["return_dict"] if inputs["return_dict"] is not None else self.config.return_dict
         if not use_cache:
-            decoder_input_ids, decoder_padding_mask, causal_mask = self._prepare_bart_decoder_inputs(
-                inputs,
-                decoder_input_ids=decoder_input_ids,
-                decoder_attn_mask=decoder_attention_mask,
+            inputs["decoder_input_ids"], decoder_padding_mask, causal_mask = self._prepare_bart_decoder_inputs(
+                inputs["input_ids"],
+                decoder_input_ids=inputs["decoder_input_ids"],
+                decoder_attn_mask=inputs["decoder_attention_mask"],
                 mask_dtype=self.shared.dtype,
             )
         else:
             decoder_padding_mask, causal_mask = None, None
-        assert (
-            isinstance(encoder_outputs, TFBaseModelOutput) or encoder_outputs is None
-        ), f"got unexpected encoder outputs type {type(encoder_outputs)}"
-        if encoder_outputs is None:
-            encoder_outputs = self.encoder(
-                input_ids=input_ids,
-                attention_mask=attention_mask,
+        # Replace by proper TF asserts
+        # assert (
```

Comment:
can we maybe uncomment those lines or replace them with tf.assert

Reply:
This assert is not anymore useful because it was needed to be sure that the encoder output format is a `TFBaseModelOutput` now, thanks to my updates it can also be a Tuple, so I removed the comment.